From 21215636ddab13f5dbcda9fef548c83c6c9557e2 Mon Sep 17 00:00:00 2001
From: Ze Tan <tanze@kylinos.cn>
Date: Mon, 10 Aug 2026 16:13:27 +0800
Subject: [PATCH 3/3] smb/server: fix missing error checks in
 ksmbd_server_init()

See the procedure below:

  ksmbd_server_init
    create_proc_sessions
      ksmbd_proc_create // fail
      return -ENOMEM
    // do not check return value
    create_proc_shares
      ksmbd_proc_create // fail
      return -ENOMEM
    // do not check return value

Check the return values and use the existing procfs cleanup path on
failure.

Signed-off-by: Ze Tan <tanze@kylinos.cn>
---
 fs/smb/server/server.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/fs/smb/server/server.c b/fs/smb/server/server.c
index 0ccd123ba418..3c318f529927 100644
--- a/fs/smb/server/server.c
+++ b/fs/smb/server/server.c
@@ -623,8 +623,13 @@ static int __init ksmbd_server_init(void)
 	ret = ksmbd_proc_init();
 	if (ret)
 		goto err_unregister;
-	create_proc_sessions();
-	create_proc_shares();
+	ret = create_proc_sessions();
+	if (ret)
+		goto err_proc_cleanup;
+
+	ret = create_proc_shares();
+	if (ret)
+		goto err_proc_cleanup;
 
 	ksmbd_server_tcp_callbacks_init();
 
-- 
2.43.0

